### PR TITLE
Add support for nested validator errors

### PR DESCRIPTION
**Pull request by josbeir**
This PR makes DocumentContext work correctly when using validation errors coming from a nested Validator setup. Also changed Document requirement to EntityInterface

### CODE REVIEW

**Comment by lorenzo** [comment]
This looks good to me. Do you think we should do something similar for the FormContext in core @markstory ?

File: src/View/Form/DocumentContext.php [hunk 1]
```diff
@@ -343,11 +344,16 @@ public function error($field)
...
         $parts = explode('.', $field);
         $entity = $this->entity($parts);
+        $errors = [];
 
         if ($entity instanceof Document) {
-            return $entity->getError(array_pop($parts));
+            $errors = $entity->getError(array_pop($parts));
+
+            if (!$errors && $this->_context['entity'] instanceof Document) {
+                $errors = array_values((array)Hash::get($this->_context['entity']->getErrors(), $field, []));
```
Comment (markstory) [inline src/View/Form/DocumentContext.php]: Would using `Hash::extract()` let you skip the array cast and `array_values` call?

Comment (josbeir) [reply src/View/Form/DocumentContext.php]: I believe it does yes, took the current line from cake's core [FormContext::error](https://github.com/cakephp/cakephp/blob/master/src/View/Form/FormContext.php#L179) method.

**Comment by josbeir** [comment]
Updated to use Hash::extract instead of Hash::get, added some extra tests to ensure errors also work correctly on multiple nested entities

**Comment by lorenzo** [comment]
Good job!

**Pull request merged**